tests/eas/cpuhp.py

```diff
@@ -0,0 +1,217 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os.path
+import random
+import json
+import sys
+import time
+
+from test import LisaTest, experiment_test
+from devlib.module.hotplug import HotplugModule as hp
+
+class _CpuHotplugTest(LisaTest):
+    """
+    "Abstract" base class for generic EAS tests under CPU hotplug stress
+
+    Subclasses should provide a .workloads member to populate the 'wloads' field
+    of the experiments_conf for the Executor. Furthermore, subclasses may
+    overwrite the .hp_stress member containing the configuration used while
+    generating the random hotplug stress sequence. 
+
+    The test will be successful if the system does not halt or kill the
+    workloads because of hotplug stress. It means that the test passes if the
+    workload completes.
+    """
+    
+    test_conf = {
+        "ftrace" : {
+            "events" : [
+                "sched_overutilized",
+                "sched_energy_diff",
+                "sched_load_avg_task",
+                "sched_load_avg_cpu",
+                "sched_migrate_task",
+                "sched_switch",
+                "cpu_frequency",
+                "cpu_idle",
+                "cpu_capacity",
+                "cpuhp_enter",
+                "cpuhp_exit",
+                "cpuhp_multi_enter",
+            ],
+        },
+        "modules": ["cgroups", "hotplug"],
```

Comment:
I think you also need to put `'tools': ['ftrace']` here to ensure trace-cmd gets installed on the target. I also think that means you won't need the explicit `te.calibration()` below.

Reply:
This has been addressed with `'tools'; ['ftrace']`